src/Hooks/UseAndroidBackNavigation.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * @Author: Erik Slovak <erik.slovak@technologystudio.sk>
+ * @Date: 2022-11-21T16:11:22+01:00
+ * @Copyright: Technology Studio
+**/
+
+import {
+  useCallback,
+  useEffect,
+} from 'react'
+import type {
+  NavigationContainerRefWithCurrent,
+} from '@react-navigation/native'
+import { BackHandler } from 'react-native'
+
+import { backHandlerManager } from '../Api/BackHandlerManager'
+
+export const useAndroidBackNavigation = (navigationRef: NavigationContainerRefWithCurrent<ReactNavigation.RootParamList>): void => {
+  const onBackPressHandler = useCallback((): boolean => {
+    const navigationState = navigationRef.getRootState()
+    if (!navigationState || navigationState.index === 0) {
+      return false
+    }
+    if (backHandlerManager.handlerList.length > 0) {
+      return backHandlerManager.handlerList.some(handler => handler())
+    }
```

Comment:
Shouldn't we navigate back if none of them handles the event, from where you have this logic ?